extensions/blocks/contact-form/components/jetpack-newsletter-integration-settings.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { BaseControl, Button, ExternalLink, PanelBody } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback, useMemo } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+
+const useInsertConsentBlock = () => {
+	const selectedBlock = useSelect( select => select( 'core/block-editor' ).getSelectedBlock(), [] );
+	const { insertBlock } = useDispatch( 'core/block-editor' );
+
+	const insertConsentBlock = useCallback( async () => {
+		const buttonBlockIndex = ( selectedBlock.innerBlocks ?? [] ).findIndex(
+			( { name } ) => name === 'jetpack/button'
+		);
+		if ( buttonBlockIndex === -1 ) {
+			return;
+		}
+
+		const newConsentBlock = await createBlock( 'jetpack/field-consent' );
+		await insertBlock( newConsentBlock, buttonBlockIndex, selectedBlock.clientId, false );
+	}, [ insertBlock, selectedBlock.clientId, selectedBlock.innerBlocks ] );
+
+	return { insertConsentBlock };
+};
+
+const NoConsentBlockSettings = () => {
+	const { insertConsentBlock } = useInsertConsentBlock();
+
+	return (
+		<>
+			<p>
+				{ __(
+					'You’re already collecting email contacts. Why not make sure you have permission to email them too?',
+					'jetpack'
+				) }
+			</p>
+			<Button isSecondary onClick={ insertConsentBlock } style={ { marginBottom: '1em' } }>
+				{ __( 'Add email permission request', 'jetpack' ) }
+			</Button>
+		</>
+	);
+};
+
+const NewsletterIntegrationSettings = () => {
+	const selectedBlock = useSelect( select => select( 'core/block-editor' ).getSelectedBlock(), [] );
+
+	const hasConsentBlock = useMemo(
+		() => selectedBlock.innerBlocks.some( ( { name } ) => name === 'jetpack/field-consent' ),
+		[ selectedBlock.innerBlocks ]
+	);
```

Comment:
What do you think about also checking if the form includes an email field? Some forms do not, and we would not want to display the invitation for a newsletter integration when that's the case.

Reply:
Updated the logic.